manager/manager.go

```diff
@@ -1247,7 +1314,9 @@ func (m *manager) watchForNewOoms() error {
 				continue
 			}
 			for _, cont := range conts {
-				atomic.AddUint64(&cont.oomEvents, 1)
+				if err := m.addOrUpdateOomInfo(cont, oomInstance.TimeOfDeath); err != nil {
+					klog.Errorf("failed to add OOM info for %q: %v", oomInstance.ContainerName, err)
+				}
```

Comment:
This is dependent on the existence of container: https://github.com/google/cadvisor/blob/0b6dfebf06a55e606935c6aad60b20b49b846bdc/manager/manager.go#L1307-L1311

If we want to address the current dependency issue of the metric on the container lifecycle, I think we shouldn't rely on the containers seen by the manager at all and rather just produce a metrics based on the info that cadvisor parses via the oom parser.

Reply:
shouldn't it be possible to track the necessary information before containers are killed to have them available independent of the container lifecycle?

Reply:
It is possible but that would require to cache all the container data.
For a separate reason, there was an attempt to create one in https://github.com/google/cadvisor/pull/2974, but I don't know if any of the maintainers would be willing to pull on the trigger on such a big change. At least it wasn't conceivable in the other PR due to the memory implications of such a change.
Also Kubernetes is and has been to trying to move away from cadvisor in favor of [container-runtime metrics](https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/2371-cri-pod-container-stats/README.md) which doesn't go in the favor of such an investment in cadvisor.

Reply:
> It is possible but that would require to cache all the container data.

true, but not _all_ container data, just the bare-minimum to be able to produce useful metrics